include/git2/sys/filter.h

```diff
@@ -139,6 +139,23 @@ GIT_EXTERN(uint32_t) git_filter_source_options(const git_filter_source *src);
  */
 
 /**
+ * Options to alter filter behavior.
+ *
+ * There are options that affect filter behavior that are intrinsic to the
+ * definition of the filter (as opposed to the `git_filter_opt_t` values
+ * that are specific to the individual application of the filter).
+ *
+ * - GIT_FILTER_DONT_PRELOAD_WORKDIR: if this filter is the first in a
+ *   filter chain, when applying with `GIT_FILTER_TO_ODB` don't load the
+ *   file contents from the working directory; instead the `apply`
+ *   callback will get an empty `git_buf` and the filter must process the
+ *   on-disk data by itself.
+ */
+typedef enum {
+	GIT_FILTER_DONT_PRELOAD_WORKDIR = (1u << 0),
```

Comment:
ProTip: you can put the description of `GIT_FILTER_DONT_PRELOAD_WORKDIR` as a doc comment on the value and docurium will show it there.